contracts/SimpleSwap.sol

```diff
@@ -121,58 +121,35 @@ contract SimpleSwap {
     _submitChequeInternal(beneficiary, serial, amount, timeout);
   }
 
-  /// @dev helper function to calculate payout value while respecting hard deposits
-  /// @param beneficiary the address to send to
-  /// @param value maximum amount to send
-  /// @return payout amount that was actually paid out
-  /// @return payout amount that bounced
-  function _computePayout(address payable beneficiary, uint value) internal returns (uint payout, uint bounced) {
-    /* part of hard deposit used */
-    payout = Math.min(value, hardDeposits[beneficiary].amount);
-    /* if there some of the hard deposit is used update the structure */
-    if(payout != 0) {
-      hardDeposits[beneficiary].amount -= payout;
-      totalDeposit -= payout;
-    }
-
-    /* amount of the cash not backed by a hard deposit */
-    uint rest = value - payout;
-    uint liquid = liquidBalance();
-
-    if(liquid >= rest) {
-      /* swap channel is solvent */
-      payout = value;
-    } else {
-      /* part of the cheque bounces */
-      payout += liquid;
-      bounced = rest - liquid;
-    }
-  }
-
   /// @notice attempt to cash latest cheque
   /// @param beneficiary beneficiary for whose cheque should be paid out
-  function cashCheque(address payable beneficiary) public {
-    ChequeInfo storage info = cheques[beneficiary];
-
+  /// @param requestPayout amount requested to pay out
+  function cashCheque(address payable beneficiary, uint requestPayout) public returns (uint) {
```

Comment:
What's the purpose of the return value? Do we expect some beneficiaries to be contracts?

Reply:
I thought I could use it in Swap.sol, but this is not the case. Can be removed for now

Reply:
I foresee that we have to add it back later, but for now we can omit it.